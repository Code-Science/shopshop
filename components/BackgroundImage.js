import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
import styles from '../styles/components/BackgroundImage.module.css';

function BackgroundImage() {
  return (
    <div className={styles.backgroundImage}>
      <Image
        src="/images/backimg.jpg"
        alt="home banner"
        objectFit="cover"
        layout="fill"
      />
      <div className={styles.shade}>
        <p className={styles.text}>
          Discover Yourself{' '}
          <Link href="/collection/women/accessories">
            <a className={styles.shopBtn}>
              <br />
              Accessories
            </a>
          </Link>
        </p>
      </div>
    </div>
  );
}

export default BackgroundImage;
