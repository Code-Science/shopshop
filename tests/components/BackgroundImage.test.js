import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import BackgroundImage from '../../components/BackgroundImage';

jest.mock('../../store/StateProvider');

configure({ adapter: new Adapter() });

describe('BackgroundImage Component', () => {
  const wrapper = shallow(<BackgroundImage />);

  it('expect to render BackgroundImage component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
