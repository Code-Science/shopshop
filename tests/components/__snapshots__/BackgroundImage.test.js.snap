// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage Component snapshot test 1`] = `
<div
  className="backgroundImage"
>
  <Image
    alt="home banner"
    layout="fill"
    objectFit="cover"
    src="/images/backimg.jpg"
  />
  <div
    className="shade"
  >
    <p
      className="text"
    >
      Discover Yourself
       
      <Link
        href="/collection/women/accessories"
      >
        <a
          className="shopBtn"
        >
          <br />
          Accessories
        </a>
      </Link>
    </p>
  </div>
</div>
`;
